// @volcengine/openapi/lib/services/tls/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultService = exports.TlsService = void 0;
const service_1 = __importDefault(require("./service"));
const SERVICE = "TLS";
// const REGION = "cn-beijing";
const VERSION = "0.2.0";
const PROTOCOL = "https:";
class TlsService extends service_1.default {
    constructor(options) {
        super(Object.assign({ version: VERSION, serviceName: SERVICE, protocol: PROTOCOL }, options));
        this.DescribeProjects = this.createAPI("DescribeProjects", {
            method: "GET",
        });
        this.DescribeProject = this.createAPI("DescribeProject", {
            method: "GET",
        });
        this.CreateProject = this.createAPI("CreateProject", {
            method: "POST",
        });
        this.ModifyProject = this.createAPI("ModifyProject", {
            method: "PUT",
        });
        this.DeleteProject = this.createAPI("DeleteProject", {
            method: "DELETE",
        });
        // log topic related interface
        // 日志主题相关接口
        this.DescribeTopics = this.createAPI("DescribeTopics", {
            method: "GET",
        });
        this.DescribeTopic = this.createAPI("DescribeTopic", {
            method: "GET",
        });
        this.CreateTopic = this.createAPI("CreateTopic", {
            method: "POST",
        });
        this.ModifyTopic = this.createAPI("ModifyTopic", {
            method: "PUT",
        });
        this.DeleteTopic = this.createAPI("DeleteTopic", {
            method: "DELETE",
        });
        // index related interface
        // 索引相关接口
        this.DescribeIndex = this.createAPI("DescribeIndex", {
            method: "GET",
        });
        this.CreateIndex = this.createAPI("CreateIndex", {
            method: "POST",
        });
        this.ModifyIndex = this.createAPI("ModifyIndex", {
            method: "PUT",
        });
        this.DeleteIndex = this.createAPI("DeleteIndex", {
            method: "DELETE",
        });
        // topic partition related interface
        // 主题分区相关接口
        this.DescribeShards = this.createAPI("DescribeShards", {
            method: "GET",
        });
        // collection and configuration related interfaces
        // 采集配置相关接口
        this.CreateRule = this.createAPI("CreateRule", {
            method: "POST",
        });
        this.DeleteRule = this.createAPI("DeleteRule", {
            method: "DELETE",
        });
        this.ModifyRule = this.createAPI("ModifyRule", {
            method: "PUT",
        });
        this.DescribeRule = this.createAPI("DescribeRule", {
            method: "GET",
        });
        this.DescribeRules = this.createAPI("DescribeRules", {
            method: "GET",
        });
        // collect and configure binding/unbinding host groups
        // 采集配置绑定/解绑机器组
        this.ApplyRuleToHostGroups = this.createAPI("ApplyRuleToHostGroups", {
            method: "PUT",
        });
        this.DeleteRuleFromHostGroups = this.createAPI("DeleteRuleFromHostGroups", {
            method: "PUT",
        });
        // host group
        // 机器组
        this.CreateHostGroup = this.createAPI("CreateHostGroup", {
            method: "POST",
        });
        this.DeleteHostGroup = this.createAPI("DeleteHostGroup", {
            method: "DELETE",
        });
        this.ModifyHostGroup = this.createAPI("ModifyHostGroup", {
            method: "PUT",
        });
        this.DescribeHostGroup = this.createAPI("DescribeHostGroup", {
            method: "GET",
        });
        this.DescribeHostGroups = this.createAPI("DescribeHostGroups", {
            method: "GET",
        });
        this.DescribeHosts = this.createAPI("DescribeHosts", {
            method: "GET",
        });
        this.DeleteHost = this.createAPI("DeleteHost", {
            method: "DELETE",
        });
        this.DescribeHostGroupRules = this.createAPI("DescribeHostGroupRules", {
            method: "GET",
        });
        // alarm
        // 告警
        this.CreateAlarm = this.createAPI("CreateAlarm", {
            method: "POST",
        });
        this.DeleteAlarm = this.createAPI("DeleteAlarm", {
            method: "DELETE",
        });
        this.DescribeAlarms = this.createAPI("DescribeAlarms", {
            method: "GET",
        });
        this.ModifyAlarm = this.createAPI("ModifyAlarm", {
            method: "PUT",
        });
        this.CreateAlarmNotifyGroup = this.createAPI("CreateAlarmNotifyGroup", {
            method: "POST",
        });
        this.DeleteAlarmNotifyGroup = this.createAPI("DeleteAlarmNotifyGroup", {
            method: "DELETE",
        });
        this.ModifyAlarmNotifyGroup = this.createAPI("ModifyAlarmNotifyGroup", {
            method: "PUT",
        });
        this.DescribeAlarmNotifyGroups = this.createAPI("DescribeAlarmNotifyGroups", {
            method: "GET",
        });
        this.PutLogs = this.createPutLogsAPI("PutLogs");
        this.SearchLogs = this.createAPI("SearchLogs", {
            method: "POST",
        });
        this.DescribeTraceInstances = this.createAPI("DescribeTraceInstances", {
            method: "GET",
        });
        this.DeleteTraceInstance = this.createAPI("DeleteTraceInstance", {
            method: "delete",
        });
        this.CreateTraceInstance = this.createAPI("CreateTraceInstance", {
            method: "POST",
        });
        this.ModifyTraceInstance = this.createAPI("ModifyTraceInstance", {
            method: "PUT",
        });
        this.DescribeTraceInstance = this.createAPI("DescribeTraceInstance", {
            method: "GET",
        });
    }
}
exports.TlsService = TlsService;
exports.defaultService = new TlsService();
